File: lib/models/visiting_timing.dart
class VisitingTiming {
  final int dayNumber;
  final String fromTime;
  final String toTime;
  final bool nextDay;
  final bool closeStatus;
  final String tableName;

  VisitingTiming({
    required this.dayNumber,
    required this.fromTime,
    required this.toTime,
    required this.nextDay,
    required this.closeStatus,
    required this.tableName,
  });

  factory VisitingTiming.fromJson(Map<String, dynamic> json) {
    return VisitingTiming(
      dayNumber: json['dayNumber'],
      fromTime: json['fromTime'],
      toTime: json['toTime'],
      nextDay: json['nextDay'] == 1,
      closeStatus: json['closeStatus'] == 1,
      tableName: json['tableName'],
    );
  }
}

File: server/admin_app/models/shop/visitingTiming.js

module.exports = (superSequelize, DataTypes) =>
{
  const VisitingTiming = superSequelize.define('VisitingTiming', {
    dayNumber: {
      type: DataTypes.INTEGER,
      field: 'dayname'
    }, // Days 0-6 (Monday to Sunday)
    fromTime: {
      type: DataTypes.TIME,
      field: 'fromtime'
    },
    toTime: {
      type: DataTypes.TIME,
      field: 'totime'
    },
    nextDay: {
      type: DataTypes.INTEGER, defaultValue: 0,
      field: 'nextday'
    }, // 1 if close time is after midnight
    closeStatus: {
      type: DataTypes.TINYINT(1), defaultValue: 1,
      field: 'close_status'
    }, // 1=Open, 0=Closed
    tableName: {
      type: DataTypes.VIRTUAL,
      get ()
      {
        return 'visiting_timings';  // This value is set manually as it's not in the database
      }
    }
  },

    {
      timestamps: false,
      tableName: "visiting_timings"
    });
  return VisitingTiming;
};

// module.exports = Admin;